barretenberg/cpp/src/barretenberg/stdlib/hash/pedersen/pedersen.test.cpp

```diff
@@ -1,35 +1,260 @@
+            right_in += fr::one();
+        }
+
+        fr_ct left = public_witness_ct(&builder, left_in);
+        fr_ct right = witness_ct(&builder, right_in);
+
+        builder.fix_witness(left.witness_index, left.get_value());
+        builder.fix_witness(right.witness_index, right.get_value());
+
+        fr_ct out = pedersen_hash::hash({ left, right });
+
+        info("num gates = ", builder.get_num_gates());
+
+        bool result = builder.check_circuit();
+        EXPECT_EQ(result, true);
+
+        fr hash_native = crypto::pedersen_hash::hash({ left.get_value(), right.get_value() });
+        EXPECT_EQ(out.get_value(), hash_native);
+    }
+
+    static void test_pedersen_edge_cases()
+    {
+        Builder builder;
+
+        fr zero_fr = fr::zero();
+        fr one_fr = fr::one();
+        fr r_minus_one_fr = fr::modulus - 1;
+        fr r_minus_two_fr = fr::modulus - 2;
+        fr r_fr = fr::modulus;
+
+        fr_ct zero = witness_ct(&builder, zero_fr);
+        fr_ct one = witness_ct(&builder, one_fr);
+        fr_ct r_minus_one = witness_ct(&builder, r_minus_one_fr);
+        fr_ct r_minus_two = witness_ct(&builder, r_minus_two_fr);
+        fr_ct r = witness_ct(&builder, r_fr);
+
+        fr_ct out_1_with_zero = pedersen_hash::hash({ zero, one });
+        fr_ct out_1_with_r = pedersen_hash::hash({ r, one });
+        fr_ct out_2 = pedersen_hash::hash({ r_minus_one, r_minus_two });
+        fr_ct out_with_zero = pedersen_hash::hash({ out_1_with_zero, out_2 });
+        fr_ct out_with_r = pedersen_hash::hash({ out_1_with_r, out_2 });
+
+        info("num gates = ", builder.get_num_gates());
+
+        bool result = builder.check_circuit();
+        EXPECT_EQ(result, true);
+
+        EXPECT_EQ(bool(out_1_with_zero.get_value() == out_1_with_r.get_value()), true);
+
+        fr hash_native_1_with_zero = crypto::pedersen_hash::hash({ zero.get_value(), one.get_value() });
+        fr hash_native_1_with_r = crypto::pedersen_hash::hash({ r.get_value(), one.get_value() });
+        fr hash_native_2 = crypto::pedersen_hash::hash({ r_minus_one.get_value(), r_minus_two.get_value() });
+        fr hash_native_with_zero = crypto::pedersen_hash::hash({ out_1_with_zero.get_value(), out_2.get_value() });
+        fr hash_native_with_r = crypto::pedersen_hash::hash({ out_1_with_r.get_value(), out_2.get_value() });
+
+        EXPECT_EQ(out_1_with_zero.get_value(), hash_native_1_with_zero);
+        EXPECT_EQ(out_1_with_r.get_value(), hash_native_1_with_r);
+        EXPECT_EQ(out_2.get_value(), hash_native_2);
+        EXPECT_EQ(out_with_zero.get_value(), hash_native_with_zero);
+        EXPECT_EQ(out_with_r.get_value(), hash_native_with_r);
+        EXPECT_EQ(hash_native_with_zero, hash_native_with_r);
+    }
+
+    static void test_pedersen_large()
+    {
+        Builder builder;
+
+        fr left_in = fr::random_element();
+        fr right_in = fr::random_element();
+        // ensure left has skew 1, right has skew 0
```

Comment:
I don't really understand the importance of this test; ie why we are testing for skewness -- It seems to be something that was only important when we were doing wnaf?